src/modules/WhatsNew/store/store.ts

```diff
@@ -0,0 +1,24 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { getUnreadMessagesCount } from './actions';
+import { MessagesState } from './types';
+
+const initialState: MessagesState = {
+  unreadCount: 0,
+};
+
+export const messagesSlice = createSlice({
+  name: 'messages',
+  initialState,
+  reducers: {},
+  extraReducers: {
+    [getUnreadMessagesCount.fulfilled.type]: (state, action: PayloadAction<number>) => {
+      state.unreadCount = action.payload;
+    },
+    [getUnreadMessagesCount.rejected.type]: (state) => {
+      state.unreadCount = 0;
```

Comment:
i am not sure that we have to change previous unreadCount to 0 if we couldn't get response, if it is a first request for message, we have an initialValue, if not we have previos value in store.

Reply:
@EvgeniyKvasyuk No we don't have to set it 0 we can show previously retrieved count. I forgot to add TODO-Need to do error handling

Reply:
and create a hook which make an request for message and return status of this request, and check inside the hook if this request is still in process then not request again.

Reply:
maybe we will add some status or process to our stores? with next values:  initial, requested, succeeded, failed what do you think?

Reply:
what will be the purpose? createAsyncThunk gives use pending state, rejected and fulfilled states the above function gets triggered on the basis of that phase we can manage loading state error state and actual payload if we wanted here based on the phase

Reply:
> what will be the purpose? createAsyncThunk gives use pending state, rejected and fulfilled states the above function gets triggered on the basis of that phase we can manage loading state error state and actual payload if we wanted here based on the phase

to get information of the status of request to avoid unnecessary requests from different part of application

Reply:
@EvgeniyKvasyuk That was for Vlad. I got your point of a centralised hook for all requests. I'll make sure I tag the person so that only they will address it to avoid further confusion.

Reply:
Got it. But I was developing Vlad`s thoughts about statuses of requests too. We can make something like here to store request status. And then we can give access to this information with a hook, what I talked about, or with a special selector  (getMessageRequestStatusSelector (for example)) - https://redux-toolkit.js.org/rtk-query/usage/migrating-to-rtk-query#thunk-action-creator -